src/test/java/uk/ac/ebi/ampt2d/accession/AccessioningServiceTest.java

```diff
@@ -0,0 +1,156 @@
+
+public class AccessioningServiceTest {
+
+    private  AccessioningService<String> service;
+
+    @Before
+    public void setUp() throws Exception {
+        AccessionGenerator<String> generator = new AccessionGenerator<String>() {
+            private static final String PREFIX = "ACC";
+
+            private int counter = 0;
+
+            @Override
+            public Map<String, String> get(Set<String> objects) {
+                Map<String, String> accessions = new HashMap<>();
+                for (String object : objects) {
+                    accessions.put(object, PREFIX + counter++);
+                }
+                return accessions;
+            }
+        };
+
+        AccessionRepository<String> repository = new AccessionRepository<String>() {
+            private Map<String, String> storage = new HashMap<>();
+
+            @Override
+            public Map<String, String> get(List<String> objects) {
+                Map<String, String> accessions = new HashMap<>();
+                for (String object : objects) {
+                    if (storage.containsKey(object)) {
+                        accessions.put(object, storage.get(object));
+                    }
+                }
+                return accessions;
+            }
+
+            @Override
+            public void add(Map<String, String> accessions) {
+                storage.putAll(accessions);
+            }
+        };
+
+        service = new AccessioningService<>(repository, generator);
+    }
+
+    @Test
+    public void everyNewObjectReceiveOneAccession() throws Exception {
+        List<String> newObjects = Arrays.asList("Object 1", "Object 2", "Object 3");
+        Map<String, String> accessions = service.getAccessions(newObjects);
+
+        assertTrue(newObjects.stream().allMatch(object -> accessions.get(object) != null));
+    }
+
+    @Test
+    public void equalsObjectsGetSameAccession() throws Exception {
+        String object1 = "Object 1";
+        String anotherObject1 = "Object 1";
+
+        List<String> newObjects = Arrays.asList(object1, anotherObject1);
+        Map<String, String> accessions = service.getAccessions(newObjects);
+
+        String accession1 = accessions.get(object1);
+        String anotherAccession1 = accessions.get(anotherObject1);
+
+        assertEquals(accession1, anotherAccession1);
+    }
+
+    @Test
+    public void differentObjectsGetDifferentAccessions() throws Exception {
+        String object1 = "Object 1";
+        String object2 = "Object 2";
+
+        List<String> newObjects = Arrays.asList(object1, object2);
+        Map<String, String> accessions = service.getAccessions(newObjects);
+
+        String accession1 = accessions.get(object1);
+        String accession2 = accessions.get(object2);
+
+        assertNotEquals(accession1, accession2);
+    }
+
+    @Test
+    public void differentCallsToTheServiceUsingTheSameObjectsWillReturnSameAccessions() throws Exception {
```

Comment:
To keep it consistent with on the methods above, maybe `multipleCallsUsing[Equal|Same]ObjectsGetSameAccessions`?